File: wup1/Java/BackAround.java
/* 
   backAround
    
   Given a string, take the last char and return 
   a new string with the last char added at the 
   front and back, so "cat" yields "tcatt". 
   The original string will be length 1 or more.
*/

public class BackAround {
    
   public static void main(String[] args) {
      BackAround ba = new BackAround();
      ba.test();
	}

   public String backAround(String str) {
      char last;
      if (str.length() > 1)
         last = str.charAt(str.length() -1);
      else
         last = str.charAt(0);

      return last + str + last;
   }

   public void test() {
      assert backAround("cat").equals("tcatt");
      assert backAround("Hello").equals("oHelloo");
      assert backAround("a").equals("aaa");
      System.out.println("All tests passed");
    }
}

File: wup1/JS/backAround.js
/*
   Warmup-1 -- backAround
   Given a string, take the last char and return a new string 
   with the last char added at the front and back, so 'cat' yields 
   'tcatt' The original string will be length 1 or more.

   Examples

   backAround('cat') → tcatt
   backAround('Hello') → oHelloo
   backAround('a') → aaa
*/

function backAround(str){
  if (str.length === 0) { return str; }
  return str[str.length -1] + str + str[str.length -1];
}

const assert = require('assert');
assert.equal(backAround('cat'), 'tcatt');
assert.equal(backAround('Hello'), 'oHelloo');
assert.equal(backAround('a'), 'aaa');
assert.equal(backAround('abc'), 'cabcc');
assert.equal(backAround('read'), 'dreadd');
assert.equal(backAround('boo'), 'obooo');
console.log("All tests passed");
